analyzers/src/SonarAnalyzer.CSharp/Trackers/CSharpArgumentTracker.cs

```diff
@@ -0,0 +1,104 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Helpers.Trackers;
+
+internal sealed class CSharpArgumentTracker : ArgumentTracker<SyntaxKind>
+{
+    protected override SyntaxKind[] TrackedSyntaxKinds =>
+        [
+            SyntaxKind.AttributeArgument,
+            SyntaxKind.Argument,
+        ];
+
+    protected override ILanguageFacade<SyntaxKind> Language => CSharpFacade.Instance;
+
+    protected override IReadOnlyCollection<SyntaxNode> ArgumentList(SyntaxNode argumentNode) =>
+        argumentNode switch
+        {
+            AttributeArgumentSyntax { Parent: AttributeArgumentListSyntax { Arguments: { } list } } => list,
+            ArgumentSyntax { Parent: BaseArgumentListSyntax { Arguments: { } list } } => list,
+            _ => null,
+        };
+
+    protected override int? Position(SyntaxNode argumentNode) =>
+        argumentNode is ArgumentSyntax { NameColon: not null }
+            or AttributeArgumentSyntax { NameColon: not null }
+            or AttributeArgumentSyntax { NameEquals: not null }
+            ? null
+            : ArgumentList(argumentNode).IndexOf(x => x == argumentNode);
+
+    protected override RefKind? ArgumentRefKind(SyntaxNode argumentNode) =>
+        argumentNode switch
+        {
+            AttributeArgumentSyntax => null,
```

Comment:
Is this case needed? It looks like it's either `ArgumentSyntax` or null.

Reply:
Technically, it is not needed, but only because the default case returns null instead of a more appropriate `throw new InvalidArgumentExcetion`. I may add a comment next to the default case or throw instead.

Reply:
Ok, a comment in this case would be great.